src/songinfo/streamdiscoverer.cpp

```diff
@@ -0,0 +1,130 @@
+#include "streamdiscoverer.h"
+
+#include <gst/pbutils/pbutils.h>
+#include "core/logging.h"
+#include "core/signalchecker.h"
+
+#include <QEventLoop>
+
+const unsigned int StreamDiscoverer::kDiscoveryTimeoutS = 10;
+
+StreamDiscoverer::StreamDiscoverer() : QObject(nullptr) {
+  // Setting up a discoverer:
+  discoverer_ = gst_discoverer_new(kDiscoveryTimeoutS * GST_SECOND, NULL);
+  if (discoverer_ == NULL) {
+    qLog(Error) << "Error creating discoverer" << endl;
+    return;
+  }
+
+  // Connecting its signals:
+  CHECKED_GCONNECT(discoverer_, "discovered", &on_discovered_cb, this);
+  CHECKED_GCONNECT(discoverer_, "finished", &on_finished_cb, this);
+
+  // Starting the discoverer process:
+  gst_discoverer_start(discoverer_);
+}
+
+StreamDiscoverer::~StreamDiscoverer() {
+  gst_discoverer_stop(discoverer_);
+  g_object_unref(discoverer_);
+}
+
+void StreamDiscoverer::discover(QString url) {
```

Comment:
Please follow the style guide: https://google.github.io/styleguide/cppguide.html

e.g. `Discover` not `discover` and `const QString& url